File: backend/src/Domain/Entities/Submission.cs
using Domain.Enums;

namespace Domain.Entities;

public class Submission
{
    public int Id {get; set;}
    
    public int AssignmentId {get; set;}
    public Assignment Assignment {get; set;} = null!;

    public int StudentId {get; set;}
    public User Student {get; set;} = null!;

    public string Answer {get; set;} = string.Empty;

    public SubmissionStatus Status {get; set;} = SubmissionStatus.Submitted;

    public decimal? Marks {get; set;}
    public string? Feedback {get; set;}

    public DateTime SubmittedAt {get; set;} = DateTime.UtcNow;
    public DateTime? UpdatedAt {get; set;}
    public DateTime? GradedAt {get; set;}
}

File: frontend/src/types/submission.ts
export enum SubmissionStatus {
  Submitted = 0,
  UnderReview = 1,
  Graded = 2,
  Returned = 3,
}


export interface Submission {
  id: number;

  assignmentId: number;
  assignmentCode: string;
  assignmentTitle: string;

  teacherId: number;

  studentId: number;
  studentName: string;
  studentEmail: string;

  answer: string;

  status: SubmissionStatus;

  marks: number | null;
  feedback: string | null;

  submittedAt: string;
  updatedAt: string | null;
  gradedAt: string | null;
}


export interface CreateSubmissionRequest {
  answer: string;
}


export interface UpdateSubmissionRequest {
  answer: string;
}


export interface GradeSubmissionRequest {
  marks: number;
  feedback?: string;
}


export interface UpdateSubmissionStatusRequest {
  status: SubmissionStatus;
}